utils/regen.go

```diff
@@ -0,0 +1,244 @@
+package utils
+
+// https://github.com/zach-klippenstein/goregen // Apache-2.0 License
```

Comment:
I think this file warrants more documentation to help readers understand its purpose